wasmplugin/plugin.go

```diff
@@ -270,14 +270,28 @@ func (ctx *httpContext) OnHttpRequestBody(bodySize int, endOfStream bool) types.
 
 func (ctx *httpContext) OnHttpResponseHeaders(numHeaders int, endOfStream bool) types.Action {
 	defer logTime("OnHttpResponseHeaders", currentTime())
-
+	tx := ctx.tx
 	if ctx.interruptionHandled {
 		// Handling the interruption (see handleInterruption) generates a HttpResponse with the required status code.
 		// If handleInterruption is raised during OnHttpRequestHeaders or OnHttpRequestBody, the crafted response is sent
 		// downstream via the filter chain, therefore OnHttpResponseHeaders is called.
-		// We expect a response that is ending the stream, with exactly one header (:status) and no body.
+		// We expect a response that is ending the stream, with a :status header equal to the status defined in the interruption
+		// and no body.
 		// See https://github.com/corazawaf/coraza-proxy-wasm/pull/126
-		if numHeaders == 1 && endOfStream {
+		status, err := proxywasm.GetHttpResponseHeader(":status")
+		if err != nil {
+			ctx.logger.Error().Err(err).Msg("Interruption already handled, failed to get :status")
+			return types.ActionPause
+		}
+		intStatus, err := strconv.Atoi(status)
+		if err != nil {
+			ctx.logger.Error().Err(err).Msg("Interruption already handled, failed to convert :status")
+			return types.ActionPause
+		}
+		// Because of extra headers that may be added alongside the :status header, this check does not scrictly enforces exactly one header
+		// See
+		// TODO: investigate if we want to remove extra headers to avoid possible leaks
```

Comment:
Remove extra headers from where?

Reply:
This PR allows having extra headers added alongside our local response generated after an interruption (eg 403 response). What I was wondering is if we want to "sanitize" and stip off all the headers, except `:status`. Still, I'm not sure about it, especially thinking about it deployed inside a mesh, maybe we expect to still have metadata headers populated.